network.py

```diff
@@ -83,8 +127,15 @@ def accuracy(y_true, y_pred):
     return K.mean(K.equal(y_true, K.cast(y_pred < 0.5, y_true.dtype)))
 
 
-def compile(model):
-    opt = RMSprop()
-    # opt = SGD(lr=0.0003, decay=1e-6, momentum=0.9, nesterov=True)
-    # opt = Nadam()
-    model.compile(loss=contrastive_loss, optimizer=opt, metrics=[accuracy])
+def compile(model, optimizer=None, loss_func=contrastive_loss):
+    allOptimizers = {
+        'sgd': SGD(lr=0.0003, decay=1e-6, momentum=0.9, nesterov=True),
+        'adam': Adam(),
+        'nadam': Nadam(),
+        'rms': RMSprop()
+    }
+
+    if optimizer in allOptimizers:
```

Comment:
Last change, assert that this is true, so we fail clearly when the optimizer is not in the list of supported ones.
```Python
assert optimizer in allOptimizers, '%s is an invalid optimizer' % optimizer
```

Reply:
Okay Sure. :) Yea. Thanks. Next time I will keep track of these.